packages/scaffold-config/src/frameworks.ts

```diff
@@ -61,6 +61,22 @@ function getBundlerDependency (bundler: WizardBundler['type'], projectPath: stri
   }
 }
 
+export const WIZARD_MOUNT_MODULES = ['cypress/react', 'cypress/react18', 'cypress/vue', 'cypress/vue2', 'cypress/angular'] as const
+
+export type WizardMountModule = typeof WIZARD_MOUNT_MODULES[number]
+
+const mountModule = (mountModule: WizardMountModule) => async (projectRoot: string) => Promise.resolve(mountModule)
+
+const reactMountModule = async (projectPath: string) => {
+  const reactPkg = await isDependencyInstalled(dependencies.WIZARD_DEPENDENCY_REACT, projectPath)
+
+  try {
+    return semver.major(reactPkg.detectedVersion!) === 18 ? 'cypress/react18' : 'cypress/react'
```

Comment:
`semver.major` can throw

Reply:
Weird, I guess it throws if the string we pass isn't valid. Instead of `!` should we just do a check using the `semver.valid` function? https://github.com/npm/node-semver#usage

```js
const vers = semver.valid(reactPkg.detectedVersion)

if (!vers) {
  throw Error(`Could not find valid version. Got ${reactPkg.version}`)
}

return semver.major(vers) === 18 ? ...
```

I like to error when things are unexpected, or at least log it - this seems better than `!`. You also won't need the `try/catch` anymore.

Reply:
Addressed in [143a9d1](https://github.com/cypress-io/cypress/pull/23251/commits/143a9d171d0389a749eac7907842e2d59390a6cf)

I don't want to throw an error here, since we do offer the `skip` button on the Install screen. Also, throwing an error during the file scaffolding would cause some weird behavior, like writing 3/4 files to disk and having the user stuck in a weird spot. I'm falling back to `cypress/react` if the dep isn't installed.